### ISSUE TITLE
[clang-tidy] `readability-redundant-typename` false negative on static data members

### ISSUE THREAD

**Issue by localspook** [llvm/llvm-project#175475]
```cpp
template <typename T>
struct S {
    typename T::R v1; // OK, warns
    static typename T::R v2; // false negative
};
```
https://godbolt.org/z/o5vbY83vY

This check is turning out to be surprisingly tricky to get right. Hopefully there aren't too many more issues to iron out...
